pandas/tests/indexing/test_partial.py

```diff
@@ -637,3 +637,66 @@ def test_partial_set_empty_frame_empty_consistencies(self):
         df.loc[0, 'x'] = 1
         expected = DataFrame(dict(x=[1], y=[np.nan]))
         tm.assert_frame_equal(df, expected, check_dtype=False)
+
+    def test_access_timezoned_datetimeindex_with_timezoned_label_utc(self):
+
+        # GH 6785
+        # timezone was ignored when string was provided as a label
+
+        first_january = pd.date_range('2016-01-01T00:00', '2016-01-01T23:59',
```

Comment:
I would really like to parametrize these to avoid the code repetition.  so i think you can do it with 2 test functions, one which slices and compares with an expected, and the 2nd function which checks for the warnings (you can actually do it with one if you add some more paramaters)

something like

```
@pytest.mark.parametrize("tz, start, end",......)
```

Reply:
I think this contradicts the idea of having the `df.iloc` check suggested by @jorisvandenbossche because that is rather specific. I would rather delete the non-naive UTC test because the CET test shows much more.

Reply:
You should be able to write the different strings so that they give the same expected frame I think